cfme/physical/physical_server.py

```diff
@@ -319,13 +320,52 @@ def find_by(self, provider, ph_name):
             if ph_name == name:
                 return self.instantiate(name=name, provider=provider or get_crud_by_name(ems_name))
 
-    def power_on(self, *physical_servers):
+    def _execute_button(self, button, option, physical_servers, handle_alert=True):
         view = self.select_entity_rows(physical_servers)
-        view.toolbar.power.item_select("Power On", handle_alert=True)
+        view.toolbar.custom_button(button).item_select(option, handle_alert=handle_alert)
+
+    def power_on(self, *physical_servers):
+        self._execute_button("Power", "Power On", physical_servers)
 
     def power_off(self, *physical_servers):
+        self._execute_button("Power", "Power Off", physical_servers)
+
+    def power_off_now(self, *physical_servers):
+        self._execute_button("Power", "Power Off Immediately", physical_servers)
+
+    def restart(self, *physical_servers):
+        self._execute_button("Power", "Restart", physical_servers)
+
+    def restart_now(self, *physical_servers):
+        self._execute_button("Power", "Restart Immediately", physical_servers)
+
+    def restart_to_sys_setup(self, *physical_servers):
+        self._execute_button("Power", "Restart to System Setup", physical_servers)
+
+    def restart_management_controller(self, *physical_servers):
+        self._execute_button("Power", "Restart Management Controller", physical_servers)
+
+    def refresh(self, provider, *physical_servers):
```

Comment:
**Required**

`BaseProvider.refresh_provider_relationships()` already provides this functionality via REST by default and via UI with varmeth.

I realize this is providing some ability to refresh multiple providers from the collection, but I would ask that we not implement this here, and instead provide this through `BaseProvider`+varmeth when `BaseProvider` is updated to inherit `BaseCollection`.  In the meantime its trivial to loop over calls to the existing `refresh_provider_relationships` method.

Reply:
So I understand, but I prefer only test the action button to refresh a collection of the physical servers and I removed this method now. Is it ok? I just using the selenium to click in the button to refresh and I'm verifying if the provider could be refreshed with success.